File: lib/models/news/news_article.dart
import 'news_priority.dart';

class NewsArticle {
  final String id;
  final String title;
  final String summary;
  final String body;
  final String? imageUrl;
  final String category;
  final DateTime publishedAt;
  final String author;
  final NewsPriority priority;

  const NewsArticle({
    required this.id,
    required this.title,
    required this.summary,
    required this.body,
    required this.imageUrl,
    required this.category,
    required this.publishedAt,
    required this.author,
    required this.priority,
  });

  factory NewsArticle.fromJson(Map<String, dynamic> json) {
    return NewsArticle(
      id: json['id'] as String,
      title: json['title'] as String,
      summary: (json['summary'] as String?) ?? '',
      body: (json['body'] as String?) ?? '',
      imageUrl: json['image_url'] as String?,
      category: (json['category'] as String?) ?? 'Force-wide',
      publishedAt: DateTime.parse(json['published_at'] as String),
      author: (json['author'] as String?) ?? 'JCF Public Affairs',
      priority: NewsPriorityWire.fromWire(json['priority'] as String?),
    );
  }
}

File: backend/JcfDocumentHub.Domain/News/NewsArticle.cs
namespace JcfDocumentHub.Domain.News;

public sealed class NewsArticle
{
    public string Id { get; set; } = string.Empty;

    public string Title { get; set; } = string.Empty;

    public string Summary { get; set; } = string.Empty;

    public string Body { get; set; } = string.Empty;

    public string ImageUrl { get; set; } = string.Empty;

    public string Category { get; set; } = string.Empty;

    public DateTime PublishedAtUtc { get; set; }

    public string Author { get; set; } = string.Empty;

    public NewsPriority Priority { get; set; }
}

public enum NewsPriority
{
    Normal,
    High,
    Urgent
}
